pkg/cloud/aws/services/ec2/routetables.go

```diff
@@ -109,6 +110,9 @@ func (s *Service) describeVpcRouteTablesBySubnet() (map[string]*ec2.RouteTable,
 	res := make(map[string]*ec2.RouteTable)
 	for _, rt := range rts {
 		for _, as := range rt.Associations {
+			if *as.Main {
```

Comment:
Is main guaranteed to be non-nil for all RouteTableAssociations? If not, then this should probably be: `if as.Main != nil && *as.Main {`

Reply:
This explains the breaking test. Thanks.